src/hepstats/hypotests/calculators/asymptotic_calculator.py

```diff
@@ -125,35 +125,37 @@ def asimov_dataset(self, poi: POI):
             print(msg)
 
             self.set_params_to_bestfit()
-
             poiparam.floating = False
 
-            with poiparam.set_value(poivalue):
-                for trial in range(5):
-                    minimum = minimizer.minimize(loss=self.loss)
-                    if minimum.valid:
-                        break
+            if not self.loss.get_params():
+                values = {poiparam: {"value": poivalue}}
+
+            else:
+                with poiparam.set_value(poivalue):
+                    for trial in range(5):
+                        minimum = minimizer.minimize(loss=self.loss)
+                        if minimum.valid:
+                            break
+                        else:
+                            # shift other parameter values to change starting point of minimization
+                            for p in self.parameters:
+                                if p != poiparam:
+                                    p.set_value(
+                                        get_value(p) * np.random.normal(1, 0.02, 1)[0]
+                                    )
                     else:
-                        # shift other parameter values to change starting point of minimization
-                        for p in self.parameters:
-                            if p != poiparam:
-                                p.set_value(
-                                    get_value(p) * np.random.normal(1, 0.02, 1)[0]
-                                )
-                else:
-                    msg = "No valid minimum was found when fitting the loss function for the alternative"
-                    msg += f"hypothesis ({poi})."
-                    warnings.warn(msg)
+                        msg = "No valid minimum was found when fitting the loss function for the alternative"
```

Comment:
Would it make sense here to mention the number of trials, which is 5?

Reply:
Yes totally.